src/instructlab/configuration.py

```diff
@@ -300,42 +352,62 @@ def get_model_family(forced, model_path):
-    ):
-        if config_file == "DEFAULT":
-            config_obj = get_default_config()
-        elif not os.path.isfile(config_file):
-            config_obj = None
-            ctx.fail(
-                f"`{config_file}` does not exists, please run `ilab init` "
-                "or point to a valid configuration file using `--config=<path>`."
-            )
-        else:
-            try:
-                config_obj = read_config(config_file)
-            except ConfigException as ex:
-                raise click.ClickException(str(ex))
-        # setup logging
-        log.configure_logging(log_level=config_obj.general.log_level.upper())
-        ctx.obj = Lab(config_obj)
-        # default_map holds a dictionary with default values for each command parameters
-        training_dict = get_dict(ctx.obj.config)
-        # since torch and train args are sep, they need to be combined into a single `train` entity for the default map
-        # this is because the flags for `ilab model train` will only be populated if the default map has a single `train` entry, not two.
-        training_dict["train"] = (
-            training_dict["train"]["train_args"]
-            | training_dict["train"]["torch_args"]
-            | training_dict["train"]["train_args"]["lora"]
-            | training_dict["train"]["train_args"]["deepspeed_options"]
+def init(ctx, config_file: str):
+    if ctx.invoked_subcommand in {"init", "sysinfo"} or "--help" in sys.argv[1:]:
+        # do nothing
+        return
+
+    ensure_storage_directories_exist()
+    if config_file == "DEFAULT":
+        config_obj = get_default_config()
+    elif not os.path.isfile(config_file):
+        config_obj = None
+        ctx.fail(
+            f"`{config_file}` does not exists, please run `ilab init` "
+            "or point to a valid configuration file using `--config=<path>`."
         )
-        ctx.default_map = training_dict
+    else:
+        try:
+            config_obj = read_config(config_file)
+        except ConfigException as ex:
+            raise click.ClickException(str(ex))
+    # setup logging
+    log.configure_logging(log_level=config_obj.general.log_level.upper())
+    ctx.obj = Lab(config_obj)
+    # default_map holds a dictionary with default values for each command parameters
+    training_dict = get_dict(ctx.obj.config)
+    # since torch and train args are sep, they need to be combined into a single `train` entity for the default map
+    # this is because the flags for `ilab model train` will only be populated if the default map has a single `train` entry, not two.
+    training_dict["train"] = (
+        training_dict["train"]["train_args"]
+        | training_dict["train"]["torch_args"]
+        | training_dict["train"]["train_args"]["lora"]
+        | training_dict["train"]["train_args"]["deepspeed_options"]
+    )
+    # default_map holds a dictionary with default values for each command parameters
+    ctx.default_map = get_dict(ctx.obj.config)
```

Comment:
this needs to be `training_dict` not `get_dict`, this is how the flags are auto populated from the cfg

Reply:
Fixed, thanks for catching this!